Builds/CMake/CMakeFuncs.cmake

```diff
@@ -147,11 +147,7 @@ macro(setup_build_cache)
     set(CMAKE_CONFIGURATION_TYPES
       DebugClassic
       ReleaseClassic)
-    if(${CMAKE_BUILD_TYPE} STREQUAL "Debug")
-      set(CMAKE_BUILD_TYPE DebugClassic)
-    elseif(${CMAKE_BUILD_TYPE} STREQUAL "Release")
-      set(CMAKE_BUILD_TYPE ReleaseClassic)
-    endif()
+    set(CMAKE_BUILD_TYPE "${CMAKE_BUILD_TYPE}Classic")
```

Comment:
The `project` statement needs to be earlier in the CMakeList.txt file (it is resetting the build type). This seems to work for gcc.release.unity (will need to test others, but I expect it works): https://github.com/seelabs/rippled/commit/8b03ec593bd01f27f6cffcbfc4ce0dad6d1886f6